source/mainnet/smart-contracts/tutorials/piggy-bank/testing.rst

```diff
@@ -231,43 +406,90 @@ Test inserting CCD into a piggy bank
 ====================================
 
 Next, you should test the different functions for interacting with a piggy bank.
-This works similarly to how you test init functions, in that you construct test
-versions of the arguments.
-For receive functions that means constructing |TestReceiveContext|_ and
-|TestHost|_, the latter of which expects the initial contract state.
+You will start by testing the ``insert`` entrypoint on an intact piggy bank contract.
 
-To test ``piggy_insert`` you also need to provide some amount of CCD:
+Create a new test case, named ``test_insert_intact``, and use the helper method ``create_chain_and_contract`` from the previous section to get a chain with two accounts and an initialized piggy bank contract.
 
 .. code-block:: rust
 
-   let ctx = TestReceiveContext::empty();
-   let host = TestHost::new(PiggyBankState::Intact, TestStateBuilder::new());
-   let amount = Amount::from_micro_ccd(100);
+   // .. Imports, constants, and other functions omitted for brevity.
+
+   #[test]
+   fn test_insert_intact() {
+       let (mut chain, initialization) = create_chain_and_contract();
+   }
+
+Note that you must mark the ``chain`` variable as mutable, since contract updates mutate it.
 
-When calling ``piggy_insert`` you get back a result with a return value as
-opposed to the initial state that you get from ``piggy_init``:
+Now, you are ready to update the contract with the |Chain_contract_update|_ method, which has parameters similar to |Chain_contract_init|_:
 
-.. code-block:: rust
+- A |Signer|_ to sign the transaction.
+- An ``invoker`` of type |AccountAddress|_, which pays for the transaction.
+- An ``sender`` of type |Address|_, which can either be an |AccountAddress|_ or a |ContractAddress|_.
+  The main utility of the parameter is to simulate calls from a contract without having to create a dummy contract the simply forwards the call.
+- A maximum |Energy|_ that the contract update can use.
+- A |ContractAddress|_, which you get from the initialization variable.
+- An |OwnedReceiveName|_, that specifies which receive name in the module you want to initialize.
+
+  - A "receive name" is the contract name concatenated with the entrypoint name and a dot in between.
+  - In this example, the contract ``my_contract`` and the entrypoint ``my_entrypoint`` combine to the receive name ``my_contract.my_entrypoint``.
+  - You construct it with either |OwnedReceiveName_new|_, which checks the format and returns a ``Result``, or |OwnedReceiveName_new_unchecked|_, which performs no checks.
 
-   let result = piggy_insert(&ctx, &host, amount);
+- An |OwnedParameter|_, which is a wrapper over a byte array that you construct with
```

Comment:
would it be appropriate to add at the end of this line "...one of the following:"